src/items/items.js

```diff
@@ -170,8 +170,45 @@ class Item {
    * @type {string|null}
    */
   get previousState () {
-    const state = this.rawItem.getLastState();
-    return state == null ? null : state.toString();
+    return this.previousRawState == null ? null : this.previousRawState.toString();
+  }
+
+  /**
+   * Numeric representation of Item previous state, or `null` if state is not numeric or not available.
+   * @type {number|null}
+   */
+  get previousNumericState () {
+    const previousState = this.previousRawState == null ? null : this.previousRawState.toString();
+    if (previousState == null) return null;
```

Comment:
Why doing an additional null check here? 
Can toString return null?

Please also have a look at special handling of Color Items, see numericState above.

Reply:
previousRawState can be null, but I will refactor that.